### PR TITLE
Graphite: Convert to use grafana-plugin-sdk-go contracts

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:
it is a part of #22918

**Which issue(s) this PR fixes**:
Ref  #36219 

**Special notes for your reviewer**: 
Take Graphite as a simple example for sdk convertion, + dataframe convertion

### CODE REVIEW

File: pkg/tsdb/graphite/graphite.go [hunk 1]
```diff
@@ -10,51 +10,141 @@ import (
...
+
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+		model := datasourceInfo{
+			BasicAuthEnabled:        settings.BasicAuthEnabled,
+			HTTPClient:              client,
+			URL:                     settings.URL,
+			JSONData:                jsonData,
+			DecryptedSecureJSONData: settings.DecryptedSecureJSONData,
```
Comment (marefr) [inline pkg/tsdb/graphite/graphite.go:89]: Would remove these and replace with a field for BasicAuthPassword.

File: pkg/tsdb/graphite/graphite.go [hunk 2]
```diff
@@ -10,51 +10,141 @@ import (
...
+
+		jsonData := map[string]interface{}{}
+		err = json.Unmarshal(settings.JSONData, &jsonData)
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
```
Comment (marefr) [inline pkg/tsdb/graphite/graphite.go:83]: Duplicate?

```suggestion
```

File: pkg/tsdb/graphite/graphite.go [hunk 3]
```diff
@@ -10,51 +10,141 @@ import (
...
-type GraphiteExecutor struct {
+const (
+	dsName = "grafana-graphite-datasource"
+)
+
+var (
+	glog = log.New("tsdb.graphite")
+)
+
+type graphiteExecutor struct {
```
Comment (marefr) [inline pkg/tsdb/graphite/graphite.go:39]: I don't think this is needed any longer since you have the `GraphiteService`

**Review by marefr — COMMENTED** [review]
Left some comments/suggestions

**Comment by ying-jeanne** [comment]
this would be merged after the dataframe convertion pr https://github.com/grafana/grafana/compare/ifrost/graphite-dataframes

File: pkg/tsdb/graphite/graphite.go [hunk 4]
```diff
@@ -10,51 +10,141 @@ import (
...
+
+	/*
+		graphite doc about from and until, with sdk we are getting absolute time
+		https://graphite-api.readthedocs.io/en/latest/api.html#from-until
+		In legacy code, we have to convert string to time.Time, so a lot extra operation, there is no need anymore
+	*/
+	from := fmt.Sprintf("%02d:%02d_%02d%02d%02d", q.TimeRange.From.Hour(), q.TimeRange.From.Minute(),
+		q.TimeRange.From.Year(), q.TimeRange.From.Month(), q.TimeRange.From.Day())
+	until := fmt.Sprintf("%02d:%02d_%02d%02d%02d", q.TimeRange.To.Hour(), q.TimeRange.To.Minute(),
+		q.TimeRange.To.Year(), q.TimeRange.To.Month(), q.TimeRange.To.Day())
```
[the remark this answers is not in the document]
Comment (ying-jeanne) [reply pkg/tsdb/graphite/graphite.go:146]: trying to use one of the way for abosolute time that doc is specified https://graphite-api.readthedocs.io/en/latest/api.html#from-until, let me check.

File: pkg/tsdb/graphite/graphite.go (hunk 1 above)
Comment (ying-jeanne) [reply pkg/tsdb/graphite/graphite.go:89]: you are right :)

**Comment by ying-jeanne** [comment]
tested with graphite datasource, seems working
Screenshot 2021-06-28 at 13 59 01
 fine

**Review by marefr — COMMENTED** [review]
Looks great :+1: Left a comment of a suggestion we did for opentsdb. I'll take this for a spin as well.

**Comment by marefr** [comment]
Tested with alerting and seems to work as expected :+1: Tested v0.9 and v1.0 with just one query and v1.1 using tags and multiple series averaging two series A and B in series C and created alert rule for C:
image

Alert notifications seems to work as expected including using of tags for v1.1 (name and aggregatedBy tags):
image

**Comment by ifrost** [comment]
> Alert notifications seems to work as expected including using of tags for v1.1 (name and aggregatedBy tags):

Oh nice! I didn't know Graphite adds `aggregatedBy` tag when an aggregation function is used :O

**Review by ifrost — APPROVED** [review]
There's just a small conflict with main. I can't review the code but I played with it a bit and everything seems to be working fine 👍

**Review by marefr — CHANGES_REQUESTED** [review]
Seems like `public/app/plugins/datasource/grafana-azure-monitor-datasource/dashboards/` changed unintentionally?

**Comment by ying-jeanne** [comment]
> Seems like `public/app/plugins/datasource/grafana-azure-monitor-datasource/dashboards/` changed unintentionally?

it is wierd, i redid the merge on main, still get the same, is that possible that we have post commit script is doing reformatting which is doing this? :D feel blind

**Review by marefr — APPROVED** [review]
LGTM great work

**Pull request merged**